src/tracking.F90

```diff
@@ -64,6 +74,18 @@ subroutine transport(p)
       call initialize_particle_track()
     endif
 
+    ! Recover particle coordinates for dd runs
+    if (dd_run .and. current_stage > 1) then
+      ! Move particle slightly forward in case particle is on a surface
+      xyz_temp = p % coord(1) % xyz
+      p % coord(1) % xyz = p % coord(1) % xyz + TINY_BIT * p % coord(1) % uvw
+      call find_cell(p, found_cell)
+      if (.not. found_cell) then
+        call fatal_error("Could not locate particle after crossing domain.")
+      end if
+      p % coord(1) % xyz = xyz_temp
+    end if
```

Comment:
I think we can delete this code block.  `find_cell` takes into account the particle's direction so it should properly handle surfaces without the need to shift by `TINY_BIT`.  I changed this block to say
```f90
    if (dd_run .and. current_stage > 1) then
      p % coord(p % n_coord) % cell = NONE
    end if
```
(so we would call `find_cell` inside the event loop) and the tests pass.  Even the above code should be unnecessary if you follow my suggestion of initializing particles before loading from buffer.